Towerfall/Assets/_Main/Scripts/Player/PlayerManager.cs

```diff
@@ -7,35 +8,61 @@
 
 namespace Towerfall.Managers
 {
-    public partial class PlayerManager : IPlayerControllerInput
+    public partial class PlayerManager : IPlayerControllerInput, IPlayerControllerEvent
     {
         [Inject] private IPlayerInput _playerInput;
         [Inject] private IPlayerProperties _playerProperties;
 
-        private Subject<float> _jumpSubject = new Subject<float>();
-        private Subject<Vector2> _moveSubject = new Subject<Vector2>();
-        
-        public IObservable<float> Jump => _jumpSubject.AsObservable();
-        public IObservable<Vector2> Move => _moveSubject.AsObservable();
+        private Subject<float> _jumpStartSubject = new Subject<float>();
+        private Subject<float> _moveSubject = new Subject<float>();
+        private Subject<Vector2> _dashStartSubject = new Subject<Vector2>();
+
+        private Subject<Unit> _dashEndSubject = new Subject<Unit>();
         
-        private void PassJumpEvent()
+        public IObservable<float> JumpStart => _jumpStartSubject.AsObservable();
+        public IObservable<float> Move => _moveSubject.AsObservable();
+        public IObservable<Vector2> DashStart => _dashStartSubject.AsObservable();
+
```

Comment:
Remove this line, is OK for them to be together.

Reply:
b76a32867a9490e2a7173cbf4ca256d92da7b71f